ctapipe/image/cleaning.py

```diff
@@ -355,3 +355,43 @@ def fact_image_cleaning(
         geom, pixels_to_keep, arrival_times, min_number_neighbors, time_limit
     )
     return pixels_to_keep
+
+
+def biggest_island(geom, image, mask):
+    """Find biggest island and filter it from the image.
+
+    This function finds the biggest island in the cleaned image and isolates it
+    for subsequent parametrization.
+
+    Parameters
+    ----------
+    geom: `ctapipe.instrument.CameraGeometry`
+        Camera geometry information.
+    image: array
+        pixel values of the calibrated image.
+    mask : array
+        A boolean mask of previously cleaned pixels.
+
+    Returns
+    -------
+
+    selected_geom : `ctapipe.instrument.CameraGeometry`
+        Camera geometry information filtered for the biggest island
+    selected_image : array
+        Pixel values of the calibrated image filtered for the biggest island.
+
+    """
+    # Finds all islands from the cleaned pixels and label them
+    num_islands, labels = number_of_islands(geom, mask)
+    labels = labels.astype("int64")  # this is temporaneus: I think it's better
```

Comment:
Why? Wat is the dtype before?

Reply:
From my tests I could see that it was float64, so then when I used numpy.argmax, it would say that it was expecting int64 instead.

Reply:
Okay, then I agree that the Change should be made in num_islands directly

Reply:
I checked, for me the dtype is int32, so no reason to change this. Where did you encounter floats?

Reply:
I tested it in a separate jupyter notebook.
I placed the first 4 prints inside _biggest_island_, just after the call to _number_of_islands_, in particular, the last one is,

`print(f"dtype of labels is {labels.dtype}")`

I then remove the line in which I forcibly convert the labels to int64.
 The output of the prints is the following:

```
There are 4 islands.
Island labels:
[1. 1. 0. 0. 2. 2. 2. 2. 0. 3. 3. 0. 0. 0. 4. 4. 4.]
dtype of labels is float64
```

Then I get the following error when trying to select the biggest islands' label,

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-10-51e0d6b9b8e0> in <module>
     21 
     22 # Apply function to test
---> 23 biggest_cam, biggest_image = biggest_island(cam, image, clean_mask)
     24 
     25 # Check if the function recovers the truth

<ipython-input-9-b8099df578f5> in biggest_island(geom, image, mask)
     72 
     73     # Get the label of the biggest island and create a new mask
---> 74     biggest_label = np.argmax(np.bincount(labels[np.where(labels>0)]))
     75     print(f"The biggest one is #{biggest_label}.")
     76     biggest_mask = labels == (biggest_label)

<__array_function__ internals> in bincount(*args, **kwargs)

TypeError: Cannot cast array data from dtype('float64') to dtype('int64') according to the rule 'safe'
```

Reply:
That's strange. What are your input values? For it always returns int32

Reply:
The ones from the unit test:

```
    # Create a simple camera
    cam = CameraGeometry.make_rectangular(17, 1)
    cam.cam_id = "test"  # this overwrites the default -1 set by the function
    # Create a simple image made of ones and zero "photoelectrons"
    image = np.array([2, 1, 1, 1, 1, 2, 2, 1, 0, 2, 1, 1, 1, 0, 2, 2, 2])
    # Let's say that we clean with a simple tailut with the following parameters
    # p = 2, b = 1, min neighbors=0
    # This cleaning would find the following islands
    clean_mask = np.zeros(cam.n_pixels).astype("bool")
    clean_mask[[0, 1]] = 1
    clean_mask[[4, 5, 6, 7]] = 2  # this is the biggest
    clean_mask[[9, 10]] = 3
    clean_mask[[14, 15, 16]] = 4
```

Then I call the function and get the error,

`biggest_cam, biggest_image = biggest_island(cam, image, clean_mask)`

Reply:
Just to clear things up, the following is a minimal example:

```
cam = CameraGeometry.make_rectangular(5, 1)
image = np.array([0,0,1,1,1])
clean_mask = np.zeros(cam.n_pixels).astype("bool")
clean_mask[[2, 3, 4]] = 1 # the only and largest island
num_islands, labels = number_of_islands(cam, clean_mask)
print(f"There are {num_islands} islands.")
print(f"Islands labelled : {labels}")
print(f"dtype of labels is {labels.dtype}")
```

which returns the following output:

```
There are 1 islands.
Islands labelled : [0. 0. 1. 1. 1.]
dtype of labels is float64
```

Reply:
To be sure, I just replicated this result using only the conda version of ctapipe 0.7.

If it can be of some help, in this case when importing _number_of_islands_ I get the following NumPy warning,

```
/Users/michele/Applications/anaconda3/envs/protopipe/lib/python3.7/site-packages/corsikaio/subblocks/dtypes.py:20: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  return np.dtype(dict(**dt))
```

Reply:
Found it. Can you add, dtype='int32' to the call to np.zeros in num islands?

Reply:
Yes, in this way the problem is solved at the very beginning.